catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -591,21 +592,45 @@ private void printPhysicsLabelOnScreen() {
 		batch.end();
 	}
 
+	private GlyphLayout getScaledFont() {
+
+		GlyphLayout layout = new GlyphLayout();
+		layout.setText(font, String.valueOf((int) virtualWidthHalf));
+
+		float landscapeHeight = (virtualHeight < virtualWidth) ? virtualHeight : virtualWidth;
+		float fontPercentage = layout.height / landscapeHeight;
+		float scaleFactor = 0.03f / fontPercentage;
+
+		layout.height = layout.height * scaleFactor;
+		layout.width = layout.width * scaleFactor;
+
+		font.getData().setScale(scaleFactor);
```

Comment:
method with side effect... so after one has called this, you have to reset the font scale factor to what it was before yourself...